bin/menmosd/src/menmosd/node/test/oldmock.rs

```diff
@@ -0,0 +1,225 @@
+use std::collections::HashMap;
+use std::sync::{
+    atomic::{AtomicU32, Ordering},
+    Arc, Mutex,
+};
+
+use anyhow::{ensure, Result};
+
+use async_trait::async_trait;
+
+use bitvec::prelude::*;
+
+use interface::{BlobInfo, RoutingConfigState};
+
+use indexer::iface::*;
+
+fn tag_to_kv(tag: &str) -> Result<(&str, &str)> {
+    let splitted: Vec<_> = tag.split('$').collect();
+    ensure!(splitted.len() == 2, "invalid kv tag");
+    Ok((splitted[0], splitted[1]))
+}
+
+#[derive(Default)]
+pub struct MockDocIdMap {
+    forward_map: Mutex<HashMap<String, u32>>,
+    backward_map: Mutex<HashMap<u32, String>>,
+    recycled_ids: Mutex<Vec<u32>>,
+    next_id: AtomicU32,
+}
+
+impl DocIdMapper for MockDocIdMap {
+    fn get_nb_of_docs(&self) -> u32 {
+        self.next_id.load(Ordering::SeqCst)
+    }
+
+    fn insert(&self, doc_id: &str) -> Result<u32> {
+        let mut fwd_guard = self.forward_map.lock().unwrap();
+        let mut rwd_guard = self.backward_map.lock().unwrap();
+        let mut recycled_guard = self.recycled_ids.lock().unwrap();
+
+        let fwd_map = &mut *fwd_guard;
+        let rwd_map = &mut *rwd_guard;
+        let recycled = &mut *recycled_guard;
+
+        if let Some(i) = fwd_map.get(doc_id) {
+            Ok(*i)
+        } else {
+            let current_id = if let Some(id) = recycled.pop() {
+                id
+            } else {
+                self.next_id.fetch_add(1, Ordering::SeqCst)
+            };
+            fwd_map.insert(String::from(doc_id), current_id);
+            rwd_map.insert(current_id, String::from(doc_id));
+            Ok(current_id)
+        }
+    }
+
+    fn lookup(&self, doc_idx: u32) -> Result<Option<String>> {
+        let rwd_guard = self.backward_map.lock().unwrap();
+        let rwd_map = &*rwd_guard;
+
+        if let Some(d) = rwd_map.get(&doc_idx) {
+            Ok(Some(d.clone()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn delete(&self, doc_id: &str) -> Result<Option<u32>> {
+        let mut fwd_guard = self.forward_map.lock().unwrap();
+        let mut rwd_guard = self.backward_map.lock().unwrap();
+        let mut recycled_guard = self.recycled_ids.lock().unwrap();
+
+        let fwd_map = &mut *fwd_guard;
+        let rwd_map = &mut *rwd_guard;
+        let recycled = &mut *recycled_guard;
+
+        if let Some(doc_idx) = fwd_map.remove(doc_id) {
+            rwd_map.remove(&doc_idx);
+            recycled.push(doc_idx);
+            Ok(Some(doc_idx))
+        } else {
+            Ok(None)
+        }
+    }
+
+    fn get_all_documents_mask(&self) -> Result<BitVec> {
+        let recycled_guard = self.recycled_ids.lock().unwrap();
+        let recycled = &*recycled_guard;
+
+        // Initialize our bitvector with 1.
```

Comment:
Why?

Reply:
Its old code that I forgot to delete, but for the record its because we're creating a bitvector that contains all documents (aka a bitvector of ones).